src/System.Linq.Parallel/tests/QueryOperators/AggregateTests.cs

```diff
@@ -0,0 +1,199 @@
+using Xunit;
+
+namespace Test
+{
+    public class AggregateTests
+    {
+        [Theory]
+        [MemberData("Ranges", (object)(new int[] { 0, 1, 2, 16 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            ParallelQuery<int> query = labelled.Item;
+            if (count == 0)
+            {
+                Assert.Throws<InvalidOperationException>(() => query.Aggregate((x, y) => x + y));
+            }
+            else
+            {
+                // The operation will overflow for long-running sizes, but that's okay:
+                // The helper is overflowing too!
+                Assert.Equal(Functions.SumRange(0, count), query.Aggregate((x, y) => x + y));
+            }
+        }
+
+        [Theory]
+        [OuterLoop]
+        [MemberData("Ranges", (object)(new int[] { 1024 * 1024, 1024 * 1024 * 4 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate_Longrunning(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            Aggregate(labelled, count);
+        }
+
+        [Theory]
+        [MemberData("Ranges", (object)(new int[] { 0, 1, 2, 16 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate_Seed(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            ParallelQuery<int> query = labelled.Item;
+            Assert.Equal(Functions.SumRange(0, count), query.Aggregate(0, (x, y) => x + y));
+        }
+
+        [Theory]
+        [OuterLoop]
+        [MemberData("Ranges", (object)(new int[] { 1024 * 1024, 1024 * 1024 * 4 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate_Seed_Longrunning(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            Aggregate_Seed(labelled, count);
+        }
+
+        [Theory]
+        [MemberData("Ranges", (object)(new int[] { 0, 1, 2, 16 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate_Result(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            ParallelQuery<int> query = labelled.Item;
+            Assert.Equal(Functions.SumRange(0, count), query.Aggregate(0, (x, y) => x + y, result => result));
+        }
+
+        [Theory]
+        [OuterLoop]
+        [MemberData("Ranges", (object)(new int[] { 1024 * 1024, 1024 * 1024 * 4 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate_Result_Longrunning(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            Aggregate_Result(labelled, count);
+        }
+
+        [Theory]
+        [MemberData("Ranges", (object)(new int[] { 0, 1, 2, 16 }), MemberType = typeof(UnorderedSources))]
+        public static void Aggregate_Accumulator(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            ParallelQuery<int> query = labelled.Item;
+            int actual = query.Aggregate(
+                0,
```

Comment:
I may be misremembering, but I think that all of the aggregate tests I've seen thus far have used 0 as a seed.  Are there any tests for other values? e.g. an accumulation that does multiplication with a seed of 1? an accumulation that accumulates into a ```List<T>``` and returns a ```List<T>``` that combines all the partial ```List<T>``` instances? etc.